modules/caddyhttp/server.go

```diff
@@ -319,7 +319,7 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		// so we can track the number of bytes read from it
 		var bodyReader *lengthReader
 		if r.Body != nil {
-			bodyReader = &lengthReader{Source: r.Body}
+			bodyReader = getLengthReader(r.Body)
```

Comment:
I think it should be safe to just do it like this:

```suggestion
			bodyReader = getLengthReader(r.Body)
			defer putLengthReader(bodyReader)
```

Because `defer` runs in reverse order on the way out, so `defer s.logRequest` will run before `defer putLengthReader`.